src/SessionManager/WebBrowser.cc

```diff
@@ -0,0 +1,135 @@
+        OpenBrowser();
+    } else {
+        spdlog::debug("WebBrowser: using default browser.");
+        OpenSystemBrowser();
+    }
+}
+
+void WebBrowser::ParseCmd() {
+    if (_cmd[_cmd.size() - 1] == '&') {
+        _cmd.pop_back();
+    }
+    const std::string wildcard = "CARTA_URL";
+    auto wildcard_pos = _cmd.find(wildcard);
+    if (wildcard_pos != std::string::npos) {
+        _cmd = fmt::format("{0}{2} {1}", _cmd.substr(0, wildcard_pos), _cmd.substr(wildcard_pos + wildcard.size(), _cmd.size()), _url);
+    } else {
+        _cmd = fmt::format("{} {}", _cmd, _url);
+    }
+#if defined(__APPLE__)
+#else
+    std::istringstream isstream(_cmd);
+    std::copy(std::istream_iterator<std::string>(isstream), std::istream_iterator<std::string>(), std::back_inserter(_args));
+    fs::path path(_args[0]);
+    if (!fs::exists(path)) {
+        path = search_path(_args[0]);
+    }
+
+    if (path.empty()) {
+        spdlog::debug("Can't find {} in PATH, please check.", _args[0]);
+    } else {
+        _path_exists = true;
+    }
+    _args[0] = path.string();
+#endif
+}
+
+void WebBrowser::OpenSystemBrowser() {
+#if defined(__APPLE__)
+    std::string cmd = "open";
+#else
+    std::string cmd = "xdg-open";
+#endif
+    cmd = fmt::format("{} \"{}\"", cmd, _url).c_str();
+    spdlog::debug("WebBrowser: Trying to launch CARTA with the default browser using: {}", cmd);
+    const int ans = std::system(cmd.c_str());
+    if (ans) {
+        _status = false;
+        _error = "WebBrowser: Failed to open the default browser automatically.";
+    }
+}
+
+void WebBrowser::OpenBrowser() {
+    spdlog::debug("WebBrowser: Trying to launch CARTA with user provided browser command: {}", _cmd);
+#if defined(__APPLE__)
+    std::string cmd = "open -a " + _cmd;
+    const int ans = std::system(cmd.c_str());
+    if (ans) {
+        _status = false;
+        _error = "WebBrowser: Failed to open the browser automatically.";
+    }
+#else
+    pid_t pid = fork();
+    if (pid == 0) {
+        setsid();
+        // double fork
+        pid_t pid2 = fork();
+        signal(SIGHUP, SIG_IGN);
+        signal(SIGPIPE, SIG_IGN);
+        if (pid2 == 0) {
+            std::vector<char*> args;
+            for (auto& arg : _args) {
+                args.push_back(const_cast<char*>(arg.c_str()));
+            }
+            args.push_back(nullptr);
+
+            // redirect output using unistd
+            int fd = open("/dev/null", O_WRONLY);
+            dup2(fd, STDOUT_FILENO);
+            dup2(fd, STDERR_FILENO);
+            close(fd);
+
+            auto result = execv(args[0], args.data());
+            if (result == -1) {
+                spdlog::debug("WebBrowser: execv failed. CARTA can't start with the requiered settings in --browser.", result);
+            }
+            _exit(1);
+        } else if (pid2 == -1) {
+            spdlog::debug("WebBrowser: Failed to fork a new process. CARTA can't start with the requiered settings in --browser.");
```

Comment:
`spdlog::warning` and typo. A warning is fine here because you're exiting the forked process

Reply:
> `spdlog::warning` and typo. A warning is fine here because you're exiting the forked process

Ok, changing it to: `WebBrowser: Failed to fork a new process. CARTA can't start with the required settings in --browser.`